### DISCUSSION TITLE
How to disable Next.js cache writing to ./next/cache for read-only root filesystem in Docker?

### DISCUSSION THREAD

**Discussion by bobbytes** [vercel/next.js#67031] [Help]
### Summary

Hi Next.js Community,

I am running a Next.js application in a Docker container with the root file system set to read-only mode. I've encountered an issue where Next.js writes to the ./next/cache directory, which isn't allowed in my current setup.

I am looking for a way to disable this caching behavior to simplify my deployment. Using a writable volume for the cache has drawbacks, especially when multiple instances write to the same cache folder. This would require managing unique folders for each container instance and cleaning them up upon deprovisioning.

Is there a configuration or a workaround to completely disable the caching in ./next/cache? Any suggestions or best practices to handle this scenario effectively would be greatly appreciated.

Thank you!

**Comment by adanperez** [comment]
Running into the same issue.  Would like to disable this as it causes consistent errors.

**Comment by hy-rf** [comment]
I added unoptimized:true under images in next config then it works

**Comment by Ho-s** [comment]
I encountered the same issue.

  **Environment:**
  - Next.js 16 (standalone output)
  - Docker with `read_only: true`
  - Non-root user (uid 1001)

  **Solution that worked for me:**
  Mounting `.next/cache` as tmpfs while keeping read-only filesystem:

  ```yaml
  services:
    app:
      read_only: true
      tmpfs:
        - /tmp
        - /app/.next/cache:uid=1001,gid=1001
```

  uid and gid should match the non-root user defined in your Dockerfile:

```Dockerfile
  RUN addgroup --system --gid 1001 nodejs
  RUN adduser --system --uid 1001 nextjs
  USER nextjs
```
  This way, both fetch cache and image optimization work without disabling anything.
